### DISCUSSION TITLE
WalletConnect V2 package version error with typescript and react.

### DISCUSSION THREAD

**Discussion by RenKoya1** [WalletConnect/walletconnect-monorepo#2876] [Developer Support] [answered]
In order to update to V2, wagmi, viem, etc. need to update the typescript version to 5.0.4 or higher, but react-scripts cannot support it.

Here is a part of package.json file.

`"@web3modal/ethereum": "^2.6.2",
  "@web3modal/react": "^2.6.2",
  "viem": "^1.2.5",
  "wagmi": "^1.3.3"
`

When I run `npm i react-scripts` I get this error message.

While resolving: react-scripts@5.0.1
 Found: typescript@5.1.6
 node_modules/typescript
   peerOptional typescript@">=5.0.4" from @wagmi/chains@1.3.0
   node_modules/@wagmi/chains
     peerOptional @wagmi/chains@">=1.3.0" from @wagmi/connectors@2.6.2
     node_modules/@wagmi/connectors
       @wagmi/connectors@"2.6.2" from @wagmi/core@1.3.3
       node_modules/@wagmi/core
         peer @wagmi/core@">=1" from @web3modal/ethereum@2.6.2
         node_modules/@web3modal/ethereum
         1 more (wagmi)
     @wagmi/chains@"1.3.0" from @wagmi/core@1.3.3
     node_modules/@wagmi/core
       peer @wagmi/core@">=1" from @web3modal/ethereum@2.6.2
       node_modules/@web3modal/ethereum
         @web3modal/ethereum@"^2.6.2" from the root project
       1 more (wagmi)
   peerOptional typescript@">=5.0.4" from @wagmi/connectors@2.6.2
   node_modules/@wagmi/connectors
     @wagmi/connectors@"2.6.2" from @wagmi/core@1.3.3
     node_modules/@wagmi/core
       peer @wagmi/core@">=1" from @web3modal/ethereum@2.6.2
       node_modules/@web3modal/ethereum
         @web3modal/ethereum@"^2.6.2" from the root project
       1 more (wagmi)
   8 more (@wagmi/core, abitype, tsutils, viem, @wagmi/chains, abitype, ...)
 
 Could not resolve dependency:
 peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
 node_modules/react-scripts
   react-scripts@"*" from the root project
 
 Conflicting peer dependency: typescript@4.9.5
 node_modules/typescript
   peerOptional typescript@"^3.2.1 || ^4" from react-scripts@5.0.1
   node_modules/react-scripts
     react-scripts@"*" from the root project

**Comment by JonathanConn** [comment, marked as answer]
You would have to update typescript or workaround this error yourself
https://github.com/facebook/create-react-app/issues/13080

Best of luck
